### DISCUSSION TITLE
Question: How to create a development environment on a Linux distro (Ubuntu)?

### DISCUSSION THREAD

**Discussion by ricardopaiva** [photoprism/photoprism#1123] [General]
Hi all,

I'm trying to setup the development environment on a VM with Ubuntu distro installed.
I've followed the instructions on the Developer Guide page [()](https://docs.photoprism.org/developer-guide/setup/) and 
I'm able to open the photoprism on Chrome, on the VM, on the url http://localhost:2342
I want to use Visual Studio Code as IDE so I'm trying to setup VSCode to run the project. I was able to setup VSCode and to open Chrome after pressing F5 (Run on Debug) on VSCode, but I was not able to debug the code. I guess that VSCode is just opening Chrome with the url indicated in the launch.json file but the code is not being deployed to the container. I'm just opening the url and using the webserver on on the photoprism docker container created in the Developer Guide, but the repository I cloned to the VM is not being deployed to the container so, is not being used.

Can you please give me some guidance on how I should setup VSCode to be able to make changes and deploy/debug it?

I'm not sure if I need to use Delve to debug and if the project is already prepared to use Delve for debugging. I've read some articles or how to install Delve but didn't understand how to integrate it with the photoprism solution.

Thanks for your help.

**Comment by lastzero** [comment]
Sorry, we have no experience with VSCode whatsoever :/

Also, I rarely use a debugger and prefer a combination of unit tests and meaningful logs instead, so I'm not a great help here either...

**Comment by ricardopaiva** [comment]
Debug with logs.. .Are you serious? :|
Anyone else?
Just other question... When I run the "./photoprism start" command on the Docker container and I open the website on the url http://localhost:2342, all I get is the page with the Photoprism logo. What am I missing?
![image](https://user-images.githubusercontent.com/4399140/109435193-32307000-7a11-11eb-8709-4f27c188bb08.png)

**Comment by lastzero** [comment]
Javascript is missing. Not built or not loaded. I don't debug at all. Using a test first approach.

**Comment by ricardopaiva** [comment]
Thanks. I was able to get it working. Tomorrow I’ll try it again and share how I accomplished that. Maybe somebody else is interested as well.
Regarding the test first approach, i understand it is a good approach but i still want to be able to debug code.
